src/domain/accounts/index.ts

```diff
@@ -22,11 +22,32 @@ export * from "./primitives"
 const UserIdRegex =
   /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i
 
+export const checkedToDeviceAccountUserId = (userId: string) => {
+  if (userId.match(/^[0-9]+:[0-9]+:[a-z]+:[0-9a-z]+$/i)) {
+    return userId as UserId
+  }
+
+  return new InvalidUserId(userId)
+}
+
 export const checkedToUserId = (userId: string): UserId | ValidationError => {
-  if (!userId.match(UserIdRegex)) {
-    return new InvalidUserId(userId)
+  // kratosUserId is a uuid v4
+  if (userId.match(UserIdRegex)) {
+    return userId as UserId
+  }
+
+  // kratosUserId is a firebase jwt subject
+  if (isDeviceId(userId)) return userId as UserId
+
+  return new InvalidUserId(userId)
+}
+
+export const isDeviceId = (userId: string) => {
```

Comment:
this way to check for deviceId is not how we typically do check. 

of this is a deviceId, then we should return a typed value. (ie: `return id as DeviceId`) so that we don't have to cast outside like we're doing [here](https://github.com/GaloyMoney/galoy/pull/2071/files#diff-651d74d3e907cd7859cee097d1d418d4d434273dc9f3392e4faabc6446eb3974R159)

Reply:
updated to use `checkedToDeviceId` here https://github.com/GaloyMoney/galoy/pull/2071/commits/1bc785dfa9c9a1ff71bef82d6bb236d7a7c0fe9a